### PR TITLE
Build with Maven 3.9.6

### PR DESCRIPTION
**Pull request by Stephan202**
This should unblock PRs #434 and #491.

Suggested commit message:
```
Build with Maven 3.9.6 (#964)

Using the `setup-maven-action` GitHub action we can both simplify the
build configuration and configure the version of Maven to use.

See https://github.com/s4u/setup-maven-action
```

### CODE REVIEW

**Review by rickie — APPROVED** [review]
Niceeee, this solution is way better 🚀 !

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -30,16 +30,12 @@ jobs:
...
-        with:
-          persist-credentials: false
-      - name: Set up JDK
-        uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93 # v4.0.0
+      - name: Check out code and set up JDK and Maven
+        uses: s4u/setup-maven-action@fa2c7e4517ed008b1f73e7e0195a9eecf5582cd4 # v1.11.0
         with:
           java-version: ${{ matrix.jdk }}
-          distribution: ${{ matrix.distribution }}
-          cache: maven
```
Comment (mohamedsamehsalah) [inline .github/workflows/build.yml:42]: Why did we drop the cache ?

Did not find [anything](https://www.google.com/search?q=codeql+analysiss+and+cache&rlz=1C5CHFA_enNL1017NL1017&oq=codeql+analysiss+and+cache&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIJCAEQIRgKGKAB0gEINDc1OWowajeoAgCwAgA&sourceid=chrome&ie=UTF-8) that says this is enabled by default 🤔

**Review by mohamedsamehsalah — APPROVED** [review]
🚀 🚀

File: .github/workflows/build.yml (hunk 1 above)
Comment (Stephan202) [reply .github/workflows/build.yml:42]: Good question. The implementation shows that the check [includes](https://github.com/s4u/setup-maven-action/blob/9337884575c7c64d6ed20d9ab3ee943658d49336/action.yml#L112-L119) a caching step, which is [enabled by default](https://github.com/s4u/setup-maven-action/blob/9337884575c7c64d6ed20d9ab3ee943658d49336/action.yml#L50-L54).

Comment (mohamedsamehsalah) [reply .github/workflows/build.yml:42]: Screenshot 2024-01-13 at 17 15 49
Sorry I was being blind 😮‍💨

**Pull request merged**